src/libaktualizr/uptane/fetcher.cc

```diff
@@ -24,7 +24,7 @@ static size_t DownloadHandler(char* contents, size_t size, size_t nmemb, void* u
   uint64_t downloaded = size * nmemb;
   auto expected = static_cast<uint64_t>(ds->target.length());
   if ((ds->downloaded_length + downloaded) > expected) {
-    return ds->downloaded_length + downloaded;  // curl will abort if return unexpected size;
+    return downloaded + 1;  // curl will abort if return unexpected size;
```

Comment:
What was the reason for changing this line? Just catching the case where `ds->downloaded_length` was somehow 0?

Reply:
Here we should return value differnt to 'downloaded', to abort download. If event calls first time, ds->downloaded_length is 0, and if variable 'downloaded' is greater then 'expected' we returns  '0 + downloaded'